### PR TITLE
Preload the header and the footer menus and language queries

### PR DESCRIPTION
**Pull request by nikomakela**
HH-252.
Refactor the menus and languages preloading so that they can be used as properties for the header Navigation-component and the footer section -component. Use a new NavigationContext-component to easily and uniformly provide the preloaded data to different NextJS pages.

Same set of changes done to every app:
1. Add the header and footer menus and the languages to global static SSR properties.
2. Provide the new static props with a NavigationContext.
3. Refactor Navigation component to use non-apollo-version of the RCHC navigation component. Navigation component can use menu and languages options that are given as props. If them are provided, skip the data fetching.
4. The FooterSection component can use menu that is given as a prop. If it is provided, skip the data fetching.

Has dependency to https://github.com/City-of-Helsinki/react-helsinki-headless-cms/pull/76

### CODE REVIEW

**Comment by nikomakela** [comment]
Hint: It's probably easier to preview this a commit by commit 😊

File: apps/events-helsinki/src/common-events/components/navigation/Navigation.tsx [hunk 1]
```diff
@@ -16,19 +22,31 @@ import styles from './navigation.module.scss';
 
 type NavigationProps = {
   page?: PageType | ArticleType;
+  menu?: Menu;
+  languages?: Language[];
 };
 
-export default function Navigation({ page }: NavigationProps) {
+export default function Navigation({ page, menu, languages }: NavigationProps) {
```
Comment (nikomakela) [inline apps/events-helsinki/src/common-events/components/navigation/Navigation.tsx:29]: It would be nice to provide this Navigation as  a common component, but there are those functions which has some application configurations.

Could be considered to be done somehow in a different PR.

File: apps/events-helsinki/src/domain/app/getEventsStaticProps.ts [hunk 2]
```diff
@@ -96,36 +84,82 @@ type GetGlobalCMSDataParams = {
...
+  });
+
+  const { data: languagesData } = await client.query({
+    query: LanguagesDocument,
+  });
+
+  client.writeQuery({
+    query: LanguagesDocument,
+    data: { ...languagesData },
+  });
```
Comment (nikomakela) [inline apps/events-helsinki/src/domain/app/getEventsStaticProps.ts:142]: These adds the content to the Apollo cache. However, I don't think we get a big advantage of this, if any, because the content is now served as component props instead of hook usage.

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr126.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr126.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr126.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr126.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr126.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr126.dev.hel.ninja** :rocket::rocket::rocket:

**Review by melniiv — APPROVED** [review]
